src/meta/src/manager/catalog/mod.rs

```diff
@@ -778,6 +778,7 @@ where
         &self,
         table_id: TableId,
         internal_table_ids: Vec<TableId>,
+        fragment_manager: FragmentManagerRef<S>,
```

Comment:
Is passing in fragment appropriate? I'm using it to fetch internal table ids from the index table fragment.

This is required from `drop_table`, since it needs to drop both the index and its internal tables.

Reply:
In my opinion, it seems suitable as there are no other means to retrieve internal tables of indexes.

Reply:
Yeah, only other solution I could think of was to maintain it in the `Index` data structure. But seems like a waste and potential source of confusion to introduce a second source of truth for `Index`'s internal tables. So I used fragment_manager instead.

Only concern is lock contention. e.g. if some other thread A acquires a write lock on fragment manager when this thread B calls drop_table and acquires the mutex lock on catalog manager. 
Then if that thread A also tries to acquire a mutex lock on the catalog manager before dropping its write lock, while thread B tries to acquire read lock on fragment manager, we have deadlock. Since neither can progress.

Currently that is not the case AFAIK, just a caveat.